### PR TITLE
Add docs for structure package

### PR DESCRIPTION
**Pull request by shadowhand**
Refs #7

### CODE REVIEW

**Comment by elazar** [comment]
`structure.md` needs to be added to `mkdocs.yml`.

File: docs/structure.md [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+[`sort`][php-sort] function will be used.
+
+**`UnorderedList`** is an implementation of a [list][wiki-list] that stores
+unordered values. The same value may appear more than once. Only an indexed array
+can be used to initialize the structure. Any value can be added.
+
+**`Set`** is an implementation of a [set][wiki-set] that stores a unique values.
+The same value will *not* appear more than once. Only an indexed array can be used
+to initialize the structure. Adding an existing value to the set will have no effect.
+A set also also be added to before or after an existing element.
```
Comment (elazar) [inline docs/structure.md]: This sentence needs typo correction / rephrasing.

**Comment by elazar** [comment]
You may want to drop a link in there referencing [immutable objects](https://en.wikipedia.org/wiki/Immutable_object), but otherwise, the content looks fine.

File: docs/structure.md [hunk 2]
```diff
@@ -0,0 +1,286 @@
...
+[`sort`][php-sort] function will be used.
+
+**`UnorderedList`** is an implementation of a [list][wiki-list] that stores
+unordered values. The same value may appear more than once. Only an indexed array
+can be used to initialize the structure. Any value can be added.
+
+**`Set`** is an implementation of a [set][wiki-set] that stores a unique values.
+The same value will *not* appear more than once. Only an indexed array can be used
+to initialize the structure. Adding an existing value to the set will have no effect.
+A set can also also be added to before or after an existing element.
```
Comment (shadowhand) [inline docs/structure.md]: @elazar fixed.

Comment (elazar) [inline docs/structure.md]: @shadowhand "also" is duplicated.

File: docs/structure.md [hunk 3]
```diff
@@ -0,0 +1,300 @@
...
+[`sort`][php-sort] function will be used.
+
+**`UnorderedList`** is an implementation of a [list][wiki-list] that stores
+unordered values. The same value may appear more than once. Only an indexed array
+can be used to initialize the structure. Any value can be added.
+
+**`Set`** is an implementation of a [set][wiki-set] that stores a unique values.
+The same value will *not* appear more than once. Only an indexed array can be used
+to initialize the structure. Adding an existing value to the set will have no effect.
+A value can also be inserted into a set before or after an existing value.
```
Comment (shadowhand) [inline docs/structure.md]: @elazar reword the whole sentence to be more clear.

File: docs/structure.md [hunk 4]
```diff
@@ -0,0 +1,300 @@
...
+using a set as a dependency chain.
+
+```php
+$set = $set->withValueBefore('tomato', 'spinach');
+
+print_r($set->toArray()); // eggs, tomato, spinach
+
+$set = $set->withValueAfter('cheese', 'eggs');
+
+print_r($set->toArray()); // eggs, cheese, tomato, spinach
```
Comment (shadowhand) [inline docs/structure.md]: Also added docs for set before/after inserting.

File: docs/structure.md (hunk 3 above)
Comment (elazar) [inline docs/structure.md]: :+1:

**Comment by elazar** [comment]
@shadowhand Add `structure.md` to `mkdocs.yml` and it should be good to go.

**Comment by shadowhand** [comment]
@elazar added to mkdocs and added a couple more headers to for easier linking.

**Comment by elazar** [comment]
:+1:

**Pull request merged**